lib/middleware.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+var middleware = require('express').Router();
+
+module.exports = function generateMiddleware(options) {
+
+  options = options || {};
+
+  options['cookie-name'] = options['cookie-name'] || 'hof-cookie-check';
+
+  middleware.get('/cookies-required', function renderCookiesRequired(req, res) {
+    res.render('cookies-required');
+  });
+
+  middleware.get('/check-cookies', function checkCookie(req, res) {
+    if (Object.keys(req.cookies).length && req.cookies[options['cookie-name']]) {
+      res.redirect(req.cookies[options['cookie-name']]);
+    } else {
+      res.redirect('/cookies-required');
```

Comment:
I hadn't even thought of that! ha. It's probably nice to do it all in the one endpoint. I shall fix it up in the morning.